### PR TITLE
Sqlfluff - Refactor and update the extension

### PR DESCRIPTION
**Pull request by RobertOstermann**
This pulls in the changes from my `sqlfluff extended` extension.

### CODE REVIEW

**Comment by alanmcruickshank** [comment]
@dorzey - would you be able to review this one? I'm afraid it's far beyond my JS-Fu.

**Comment by alanmcruickshank** [comment]
@RobertOstermann - if he's not able to review then I'll do my best in a few days time. Thanks for taking the time to put this together 🙏

File: test/.sqlfluff [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[sqlfluff]
+dialect = postgres
+
+[sqlfluff:rules:L010]
+capitalisation_policy = upper
+
+[sqlfluff:rules:L014]
+extended_capitalisation_policy = lower
```
Comment (tunetheweb) [inline test/.sqlfluff:8]: Nit, missing end of file newline

```suggestion
extended_capitalisation_policy = lower

```

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Don't know VS code extensions, my JS knowledge is weak, and my TS knowledge is non-existent :-)

But did spot a couple of minor things.

**Comment by RobertOstermann** [comment]
@tunetheweb I have made the changes you requested. I also noticed a few more problems when looking through my code and testing the extension. I have fixed those issues in recent commits.

**Pull request merged**